### PR TITLE
Add golangci-lint as a new make target

### PR DESCRIPTION
**Pull request by JoelSpeed**
I would like to make sure that we uphold strong code quality within this repository, as such, I think we should be running golangci-lint over the code to keep us honest (eg having comments etc). I will set this up as a check on PRs.

### CODE REVIEW

File: pkg/controllers/controlplanemachineset/suite_test.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 	//+kubebuilder:scaffold:imports
 )
 
-var cfg *rest.Config
+var cfg *rest.Config //nolint:deadcode,unused
```
Comment (JoelSpeed) [inline pkg/controllers/controlplanemachineset/suite_test.go:34]: This was the only linter issue reported, this variable will be used in a future PR when we set up the tests, will remove the nolint directive then

**Review by damdo — COMMENTED** [review]
Nice.
/lgtm
Optional idea: how about adding a [`.golangci.yml`](https://github.com/golangci/golangci-lint/blob/master/.golangci.yml) already so we can gradually add what sorts of checks we would like for the linter?

**Comment by lobziik** [comment]
```
go run /go/src/github.com/openshift/cluster-control-plane-machine-set-operator/vendor/github.com/golangci/golangci-lint/cmd/golangci-lint run ./...
2022/02/[28](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-control-plane-machine-set-operator/6/pull-ci-openshift-cluster-control-plane-machine-set-operator-main-lint/1498391311282081792#1:build-log.txt%3A28) 20:16:53 failed to initialize build cache at /.cache/golangci-lint: mkdir /.cache: permission denied
exit status 1
make: *** [lint] Error 1
```


Suggest to point cache dir to `/tmp/` for ci runs

**Review by alexander-demichev — APPROVED** [review]
/approve

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws
/override ci/prow/e2e-aws-serial
/override ci/prow/e2e-aws-upgrade
/skip

This project isn't in payload yet so these tests aren't actually testing these changes, also, this doesn't affect the product as it's just linting

**Pull request merged**